### PR TITLE
[DX-1151] Edp 1.8.4 docs

### PR DESCRIPTION
**Pull request by slega**
### For internal users - Please add a Jira DX PR ticket to the subject!
https://tyktech.atlassian.net/browse/TT-11278
https://tyktech.atlassian.net/browse/DX-1151
<br>

---

### Preview Link
https://deploy-preview-4187--tyk-docs.netlify.app/docs/nightly/product-stack/tyk-enterprise-developer-portal/release-notes/portal-1.8.4/
<br>

### Description
- Release notes for 1.8.4.
- Updated OAS for the portal.
- Update documentation for the email customization section.
- Update the env vars configuration section.
- Update documentation for the custom attributes for the User model.
<br>

#### Screenshots (if appropriate)
<br>

## Checklist

- [ ] I have added a **preview link** to the PR description.
- [x] I have [reviewed the guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING.md) for contributing to this repository.
- [x] I have [read the technical guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING-TECHNICAL-GUIDE.md) for contributing to this repository.
- [x] Make sure you have started *your change* off *our latest `master`*.
- [x] I **labelled** the PR

___

## **Type**
documentation, enhancement

___

## **Description**
- Updated Swagger definitions to reflect new error handling and markdown content support.
- Introduced new configuration options and clarified existing settings in the documentation.
- Added comprehensive documentation for customizing the User model, including handling of PII data.
- Released detailed notes for EDP 1.8.4, highlighting new features, changes, and fixes.
- Updated email customization documentation to include new events and behaviors.
- Clarified theme upload file size limits and made minor formatting corrections.
- Reorganized documentation menu to improve navigation and accessibility.

___

## **Changes walkthrough**
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>enterprise-developer-portal-swagger.yaml</strong><dd><code>Update Swagger Definitions for EDP 1.8.4</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/assets/others/enterprise-developer-portal-swagger.yaml
<li>Updated references from <code>Status</code> to <code>OnlyErrors</code> for error schemas across <br>various paths.<br> <li> Added <code>force</code> query parameter to user deletion endpoint to force <br>deletion despite errors from credential providers.<br> <li> Introduced <code>MarkdownContent</code> and <code>MarkdownEnabled</code> fields in <code>ContentBlock</code> <br>schema for supporting markdown content.

</details>
    

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/4187/files#diff-015c718543dbd3e884dc650ed9bec9be3628cf8db439061e81b4ebf914a68295">+44/-42</a>&nbsp; </td>
</tr>                    
</table></details></td></tr><tr><td><strong>Documentation</strong></td><td><details><summary>8 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>configuration.md</strong><dd><code>Configuration Documentation Updates for EDP 1.8.4</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/content/product-stack/tyk-enterprise-developer-portal/deploy/configuration.md
<li>Updated <code>PORTAL_MAX_UPLOAD_SIZE</code> documentation to note individual file <br>size limit.<br> <li> Added <code>PORTAL_SESSION_SAMESITE</code> configuration option for controlling the <br><code>SameSite</code> cookie attribute.

</details>
    

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/4187/files#diff-bff942701e76bae5b357af2a798c59862dff033909d36d1d2bba2d878add91dd">+18/-3</a>&nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>customize-sign-up-form.md</strong><dd><code>Clarification on Custom Attributes and PII Data</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/content/product-stack/tyk-enterprise-developer-portal/getting-started/customize-sign-up-form.md
<li>Clarified handling of PII data in custom attributes for the User <br>model.

</details>
    

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/4187/files#diff-f7064ffa3eba733b5177cd75aa0c664c980bb529b3c03445694d4202d444255e">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>customise-user-model.md</strong><dd><code>New Documentation on Customizing the User Model</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/content/product-stack/tyk-enterprise-developer-portal/portal-customisation/customise-user-model.md
<li>Introduced documentation for customizing the User model in the <br>Enterprise Developer Portal.<br> <li> Detailed steps for adding custom attributes to the User model.<br> <li> Explained the significance of default attributes and their inclusion <br>in credential metadata.

</details>
    

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/4187/files#diff-4dc2577188a04db59b6e62590132cadadeb363bfa7b5848e7bea8810c1d51a33">+84/-0</a>&nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>portal-1.8.4.md</strong><dd><code>Release Notes for EDP 1.8.4</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/content/product-stack/tyk-enterprise-developer-portal/release-notes/portal-1.8.4.md
<li>Added release notes for Tyk Enterprise Developer Portal v1.8.4.<br> <li> Highlighted new features, changes, and fixes in the 1.8.4 release.

</details>
    

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/4187/files#diff-dbe9409649e688cdc43f0c171ffc67b7395d96206549d01baea85226c30d6551">+89/-0</a>&nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>email-customization.md</strong><dd><code>Email Customization Documentation Updates</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/content/tyk-stack/tyk-developer-portal/enterprise-developer-portal/customise-enterprise-portal/full-customisation/email-customization.md
<li>Updated the list of email notifications with new events and <br>recipients.<br> <li> Added details on the behavior of welcome and activation emails.

</details>
    

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/4187/files#diff-eb62c7b5890d97e76b7b0b54ba4a34343c8fe8b6b9cd41bd59249a899f9b8da6">+23/-10</a>&nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>file-structure-concepts.md</strong><dd><code>Theme Upload File Size Limit Clarification</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/content/tyk-stack/tyk-developer-portal/enterprise-developer-portal/customise-enterprise-portal/full-customisation/file-structure-concepts.md
- Noted the individual file size limit when uploading themes.

</details>
    

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/4187/files#diff-4bc74d1810243d2956e58ef1bfd084b50a2241093d8985e8afc01663678253f8">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>quick-customisation.md</strong><dd><code>Minor Formatting Correction in Quick Customisation Guide</code>&nbsp; </dd></summary>
<hr>

tyk-docs/content/tyk-stack/tyk-developer-portal/enterprise-developer-portal/customise-enterprise-portal/quick-customisation.md
- Minor formatting correction for note on image path customization.

</details>
    

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/4187/files#diff-fe06324787add05c4097f99b84c129b82b0870aebd835ff42579f318d01e5dac">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>menu.yaml</strong><dd><code>Menu Updates for New Documentation and Release Notes</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/data/menu.yaml
<li>Added new menu entry for EDP 1.8.4 release notes.<br> <li> Reorganized portal customization section to include new documentation <br>on customizing data models.

</details>
    

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/4187/files#diff-a6789c66bfa0660f7fdbe4796002ab4777677bae59a0276689c840100ff0b0b2">+48/-40</a>&nbsp; </td>
</tr>                    
</table></details></td></tr></tr></tbody></table>

___

> ✨ **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

**Comment by dcs3spp** [comment]
@slega Could you raise a Jira ticket in the DX space so that this can be reviewed and add the ticket ID in the PR title

File: tyk-docs/assets/others/enterprise-developer-portal-swagger.yaml [hunk 1]
```diff
@@ -2506,6 +2506,13 @@ paths:
           schema:
             type: integer
             example: 1
+        - description: If set to true, will force user deletion of the user and all their applications and credentials and ignore any errors from the credential providers such as OAuth2.0 IdPs and Tyk Dashboard.
```
Comment (dcs3spp) [inline tyk-docs/assets/others/enterprise-developer-portal-swagger.yaml]:
```suggestion
If set to true, will force deletion of the user with all their applications and credentials and ignore any errors from the credential providers such as OAuth2.0 IDPs and Tyk Dashboard.
```

File: tyk-docs/assets/others/enterprise-developer-portal-swagger.yaml [hunk 2]
```diff
@@ -3657,17 +3664,14 @@ components:
...
-          example: HeaderDescription
-      type: object
-    ContentBlock-create:
-      properties:
-        Content:
+        MarkdownContent:
           type: string
-          description: Content of this content block
-          example: "<p>Everything would be better connected, but the internet is messy, imperfect and unpredictable.</p>\n\t\t\t\t<p>Our engineers are dedicated to making it simple, fast and affordable to connect every system in the world.</p>\n\t\t\t\t<p>We ensure you can trust our systems to look after yours.</p>"
+          description: Markdown version of content for this content block
```
Comment (dcs3spp) [inline tyk-docs/assets/others/enterprise-developer-portal-swagger.yaml]:
```suggestion
          description: Markdown content for this content block
```

File: tyk-docs/assets/others/enterprise-developer-portal-swagger.yaml [hunk 3]
```diff
@@ -3657,17 +3664,14 @@ components:
...
+          example: "# Heading string"
+        MarkdownEnabled:
+          type: boolean
+          description: If true, the portal will display the markdown version of content of this content block.
```
Comment (dcs3spp) [inline tyk-docs/assets/others/enterprise-developer-portal-swagger.yaml]:
```suggestion
          description: If true, the portal will display the markdown for this content block.
```

File: tyk-docs/content/product-stack/tyk-enterprise-developer-portal/deploy/configuration.md [hunk 4]
```diff
@@ -146,14 +148,27 @@ This section explains how to configure session management for the portal. Using
...
-This is a security measure to prevent XSS attacks. We recommend setting it to `true` in production environments. The default value is `true`.
+**Description**: Controls whether the portal adds the `HttpOnly` attribute to the `Set-Cookie` header in all responses from the portal's backend, except for the admin APIs. This cookie attribute controls if the cookie is only accessible at the server and not by Javascript on the client side.
+This is a security measure to prevent XSS attacks.
+
+We recommend setting it to `true` in production environments. The default value is `true`.
+
+### PORTAL_SESSION_SAMESITE
+**Config file:** Session.SameSite <br/>
+**Type:** `string` <br/>
+**Description**: Controls the value of the `SameStie` attribute for the portal’s cookie. The portal adds the `SameStie` attribute with the value specified in `PORTAL_SESSION_SAMESITE` to the `Set-Cookie` header in all responses from the portal's backend, except for the admin APIs.
```
Comment (dcs3spp) [inline tyk-docs/content/product-stack/tyk-enterprise-developer-portal/deploy/configuration.md]:
```suggestion
**Description**: Controls the value of the `SameSite` attribute for the portal’s cookie. The portal adds the `SameStie` attribute with the value specified in `PORTAL_SESSION_SAMESITE` to the `Set-Cookie` header in all responses from the portal's backend, except for the admin APIs.
```

File: tyk-docs/content/product-stack/tyk-enterprise-developer-portal/portal-customisation/customise-user-model.md [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+{{< note success >}}
+**Tyk Enterprise Developer Portal**
+
+If you are interested in getting access, contact us at [support@tyk.io](<mailto:support@tyk.io?subject=Tyk Enterprise Portal Beta>)
+
+{{< /note >}}
+
+## Introduction
+
+In this section, you will learn how to customize the User model and the sign-up form for your API consumers.
```
Comment (dcs3spp) [inline tyk-docs/content/product-stack/tyk-enterprise-developer-portal/portal-customisation/customise-user-model.md]:
```suggestion
In this section, you will learn how to customise the User model and the sign-up form for your API consumers.
```

File: tyk-docs/content/product-stack/tyk-enterprise-developer-portal/portal-customisation/customise-user-model.md [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+Customizing the User model enables the storage of custom data attributes in the User profile.
```
Comment (dcs3spp) [inline tyk-docs/content/product-stack/tyk-enterprise-developer-portal/portal-customisation/customise-user-model.md]:
```suggestion
Customising the User model enables the storage of custom data attributes in the User profile.
```

File: tyk-docs/content/product-stack/tyk-enterprise-developer-portal/portal-customisation/customise-user-model.md [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+Additionally, it allows these attributes to be optionally included in the credentials metadata (therefore accessible by the gateway runtime) and exposed in the user sign-up form.
+
+This feature enables the implementation of complex business logic when processing API requests.
+For example, it is particularly useful when the quota for API calls needs to be distributed among all developers of consumer organizations.
```
Comment (dcs3spp) [inline tyk-docs/content/product-stack/tyk-enterprise-developer-portal/portal-customisation/customise-user-model.md]:
```suggestion
For example, it is particularly useful when the quota for API calls needs to be distributed among all developers of consumer organisations.
```

File: tyk-docs/content/product-stack/tyk-enterprise-developer-portal/portal-customisation/customise-user-model.md [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+In such cases, both the quota and the rate limit should be applied at the organization level, rather than according to individual credentials.
+In this event, the organisation ID should be known to the gateway in runtime. This feature helps to achieve that.
+
+## Add new Custom Attributes to the User model
+
+To customise the User model by adding new data attributes to it, navigate to the **Custom attributes** menu and the select the **User** model. Currently, it is possible to extend only the User model. In future releases we will add the same capabilities to other models.
```
Comment (dcs3spp) [inline tyk-docs/content/product-stack/tyk-enterprise-developer-portal/portal-customisation/customise-user-model.md]:
```suggestion
To customise the User model by adding new data attributes to it, navigate to the **Custom attributes** menu and then select the **User** model. Currently, it is possible to extend only the User model. In future releases we will add the same capabilities to other models.
```

File: tyk-docs/content/product-stack/tyk-enterprise-developer-portal/portal-customisation/customise-user-model.md [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+|-----------------|---------------------------------------|-----------------------------------------------------------------------------------|
+| Developer ID    | DeveloperID                           | ID of the developer who created the credential                                    |
+| Application ID  | ApplicationID                         | ID of the application to which it belongs                                         |
+| Organisation ID | OrganisationID                        | ID of the organisation to which the developer who created the application belongs |
+| Team IDs        | TeamIDs                               | Array of team IDs to which the developer, who created the application, belongs    |
+
+Additionally, it is possible to include other default attributes of the User model in the credential metadata fields.
+However, it is important to remember that metadata at the credential level will be accessible both in the gateway runtime and in the gateway database.
+Exercise caution when dealing with personally identifiable information (PII). The available default attributes include:
```
Comment (dcs3spp) [inline tyk-docs/content/product-stack/tyk-enterprise-developer-portal/portal-customisation/customise-user-model.md]:
```suggestion
Exercise caution when dealing with personally identifiable information (PII). Additional default attributes include:
```

File: tyk-docs/content/product-stack/tyk-enterprise-developer-portal/portal-customisation/customise-user-model.md [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+| Attribute         | Name of the credential metadata field | Description                                                                         |
+|-------------------|---------------------------------------|-------------------------------------------------------------------------------------|
+| First name        | First                                 | First name of the developer who created the credential                              |
+| Last name         | Last                                  | Last name of the developer who created the credential                               |
+| Email             | Email                                 | Email name of the developer who created the credential                              |
+| Role              | Role                                  | Array of team IDs to which the developer, who created the application, belongs      |
+| Organisation name | Organisation                          | Name of the organisation to which the developer who created the application belongs |
+| Teams name        | TeamNames                             | Array of team name to which the developer, who created the application, belongs     |
```
Comment (dcs3spp) [inline tyk-docs/content/product-stack/tyk-enterprise-developer-portal/portal-customisation/customise-user-model.md]:
```suggestion
| Teams name        | TeamNames                             | Array of team names to which the developer, who created the application, belongs     |
```

File: tyk-docs/content/product-stack/tyk-enterprise-developer-portal/release-notes/portal-1.8.4.md [hunk 11]
```diff
@@ -0,0 +1,89 @@
...
+## Download
+- [Docker image to pull](https://hub.docker.com/layers/tykio/portal/v1.8.3/images/sha256-3693065546348105a693a1ed5402c93bfecd480c900e1efea4a6dea674263df3?context=explore)
+- [The default theme package](https://github.com/TykTechnologies/portal-themes/blob/main/v1.8.3/default.zip)
+
+## Changelog
+#### Added
+- Added [new configuration option]({{< ref "product-stack/tyk-enterprise-developer-portal/api-documentation/tyk-edp-api" >}}) for setting the `SameSite` attribute on the portal's cookie.
+- Added [new welcome email for admin users]({{< ref "product-stack/tyk-enterprise-developer-portal/api-documentation/tyk-edp-api" >}}) that is sent when new admin account is created.
+- Added [new welcome email for developers]({{< ref "product-stack/tyk-enterprise-developer-portal/api-documentation/tyk-edp-api" >}}) that is sent when new developer account is created.
+- Added a fallback mechanism for referencing assets that searches for assets, such as images referenced in the rich text editor or markdown editor, in the container's filesystem whenever the portal can't find the referenced asset in the `PORTAL_STORAGE`.
```
Comment (dcs3spp) [inline tyk-docs/content/product-stack/tyk-enterprise-developer-portal/release-notes/portal-1.8.4.md]: Does the fallback mechanism autofill potential asset references and does it allow the user to check and validate the suggested references it has autofilled?

```suggestion
- Added a fallback mechanism for referencing assets. This searches for assets, such as images, referenced in the rich text editor or markdown editor. It searches the container's filesystem whenever the portal can't find the referenced asset in the `PORTAL_STORAGE`.
```

File: tyk-docs/content/product-stack/tyk-enterprise-developer-portal/release-notes/portal-1.8.4.md [hunk 12]
```diff
@@ -0,0 +1,89 @@
...
+
+#### Changed
+- Renamed subject for the new developer registration request from "Registering email - subject" to "Developer registration approval request - subject".
```
Comment (dcs3spp) [inline tyk-docs/content/product-stack/tyk-enterprise-developer-portal/release-notes/portal-1.8.4.md]: Explain briefly why this was done

File: tyk-docs/content/product-stack/tyk-enterprise-developer-portal/release-notes/portal-1.8.4.md [hunk 13]
```diff
@@ -0,0 +1,89 @@
...
+- Added warning regarding potential PII exposure to the [custom attributes menu]({{< ref "/product-stack/tyk-enterprise-developer-portal/portal-customisation/customise-user-model#default-attributes" >}}).
+- Changed the behavior of the portal for 404 errors. Now whenever a user requests non-existing page both private (e.i. requiring sign-in to access) or public, the portal now always renders the `not_found.tmpl` template.
+- Changed the behavior of the `Secure` cookie attribute that is set by [PORTAL_SESSION_SECURE]({{ ref "/product-stack/tyk-enterprise-developer-portal/deploy/configuration/#portal_session_secure" }}) so that the `Secure` attribute is always add to the `Set-Cookie` header whenever `PORTAL_SESSION_SECURE` is set to `true` or when TLS is enabled.
+- Changed the behavior of the developers UI in the admin app. Now, when an admin tries to remove a developer profile and some of their credentials have been removed from the credentials provider, or if the provider itself is down or unreachable, the portal asks the admin if they still want to remove the developer profile by displaying a modal window.
```
Comment (dcs3spp) [inline tyk-docs/content/product-stack/tyk-enterprise-developer-portal/release-notes/portal-1.8.4.md]:
```suggestion
- Changed the behaviour of removing a developer profile within the developers UI in the admin app. Now, when an admin tries to remove a developer profile and some of their credentials have been removed from the credentials provider, or if the provider itself is down or unreachable, the portal asks the admin if they still want to remove the developer profile by displaying a modal window.
```

File: tyk-docs/content/product-stack/tyk-enterprise-developer-portal/release-notes/portal-1.8.4.md [hunk 14]
```diff
@@ -0,0 +1,89 @@
...
+- Extended the `DELETE /users/:id` API endpoint by adding the [?force]({{< ref "/product-stack/tyk-enterprise-developer-portal/api-documentation/tyk-edp-api" >}}) parameter to remove a user anyway even if some of their credentials have been removed from the credentials provider, or if the provider itself is down or unreachable.
```
Comment (dcs3spp) [inline tyk-docs/content/product-stack/tyk-enterprise-developer-portal/release-notes/portal-1.8.4.md]:
```suggestion
- Extended the `DELETE /users/:id` API endpoint by adding the [?force]({{< ref "/product-stack/tyk-enterprise-developer-portal/api-documentation/tyk-edp-api" >}}) parameter to force removal of a user even if some of their credentials have been removed from the credentials provider, or if the provider itself is down or unreachable.
```

File: tyk-docs/content/product-stack/tyk-enterprise-developer-portal/release-notes/portal-1.8.4.md [hunk 15]
```diff
@@ -0,0 +1,89 @@
...
+- Extended the `GET /pages/:id/content-blocks/:id:` API endpoint by adding additional fields in the response body: `Content`, `MarkdownContent`, `MarkdownEnabled`, `Name`, and `PageID`.
+- Extended [filesize limit]({{< ref "/tyk-stack/tyk-developer-portal/enterprise-developer-portal/customise-enterprise-portal/full-customisation/file-structure-concepts#part-1-create-a-new-theme" >}}) for individual files in themes to 5 MB.
```
Comment (dcs3spp) [inline tyk-docs/content/product-stack/tyk-enterprise-developer-portal/release-notes/portal-1.8.4.md]:
```suggestion
- Extended [filesize limit]({{< ref "tyk-stack/tyk-developer-portal/enterprise-developer-portal/customise-enterprise-portal/full-customisation/file-structure-concepts#part-1-create-a-new-theme" >}}) for individual files in themes to 5 MB.
```

File: tyk-docs/content/product-stack/tyk-enterprise-developer-portal/release-notes/portal-1.8.4.md [hunk 16]
```diff
@@ -0,0 +1,89 @@
...
+- Made the organisation invite email's subject configurable via [the emails settings section]({{< ref "/tyk-stack/tyk-developer-portal/enterprise-developer-portal/customise-enterprise-portal/full-customisation/email-customization#list-of-email-notifications" >}}).
```
Comment (dcs3spp) [inline tyk-docs/content/product-stack/tyk-enterprise-developer-portal/release-notes/portal-1.8.4.md]:
```suggestion
- Made the organisation invite email's subject configurable via [the emails settings section]({{< ref "tyk-stack/tyk-developer-portal/enterprise-developer-portal/customise-enterprise-portal/full-customisation/email-customization#list-of-email-notifications" >}}).
```

File: tyk-docs/content/tyk-stack/tyk-developer-portal/enterprise-developer-portal/customise-enterprise-portal/full-customisation/file-structure-concepts.md [hunk 17]
```diff
@@ -147,7 +147,7 @@ Follow the example below to create a new theme called "TestTheme" using the defa
    {{< img src="/img/dashboard/portal-management/enterprise-portal/rename-a-theme.png" alt="Rename theme" >}}
 3. You can also modify other assets in the theme as described later in this guide. Once all modifications are done, you need to zip the theme and upload it to the portal.
    {{< img src="/img/dashboard/portal-management/enterprise-portal/compress-a-theme.png" alt="Zip theme" >}}
-4. To upload the theme as an admin user, navigate to **Themes** and click on the **Add new theme** button.
+4. To upload the theme as an admin user, navigate to **Themes** and click on the **Add new theme** button. Please note that the size of individual files should not exceed 5 MB and the total size of all files in the theme should not exceed `PORTAL_MAX_UPLOAD_SIZE`. This parameter is [configurable]({{< ref "/product-stack/tyk-enterprise-developer-portal/deploy/configuration#portal_max_upload_size" >}}).
```
Comment (dcs3spp) [inline tyk-docs/content/tyk-stack/tyk-developer-portal/enterprise-developer-portal/customise-enterprise-portal/full-customisation/file-structure-concepts.md]:
```suggestion
4. To upload the theme as an admin user, navigate to **Themes** and click on the **Add new theme** button. Please note that the size of individual files should not exceed 5 MB and the total size of all files in the theme should not exceed `PORTAL_MAX_UPLOAD_SIZE`. This parameter is [configurable]({{< ref "product-stack/tyk-enterprise-developer-portal/deploy/configuration#portal_max_upload_size" >}}).
```

**Review by dcs3spp — CHANGES_REQUESTED** [review]
PR LGTM, have made minor suggestions. Could you also arrange a peer review

File: tyk-docs/content/product-stack/tyk-enterprise-developer-portal/release-notes/portal-1.8.4.md [hunk 18]
```diff
@@ -0,0 +1,89 @@
...
+| /auth/password/login         | Developer portal login |
+| /auth/password/new           | Password reset         |
+- Changed the credential provisioning flow to automatically include DeveloperID, OrganisationID, ApplicationID, and TeamIDs in [the credential metadata]({{< ref "/product-stack/tyk-enterprise-developer-portal/portal-customisation/customise-user-model#default-attributes" >}}).
+- Added warning regarding potential PII exposure to the [custom attributes menu]({{< ref "/product-stack/tyk-enterprise-developer-portal/portal-customisation/customise-user-model#default-attributes" >}}).
+- Changed the behavior of the portal for 404 errors. Now whenever a user requests non-existing page both private (e.i. requiring sign-in to access) or public, the portal now always renders the `not_found.tmpl` template.
+- Changed the behavior of the `Secure` cookie attribute that is set by [PORTAL_SESSION_SECURE]({{ ref "/product-stack/tyk-enterprise-developer-portal/deploy/configuration/#portal_session_secure" }}) so that the `Secure` attribute is always add to the `Set-Cookie` header whenever `PORTAL_SESSION_SECURE` is set to `true` or when TLS is enabled.
```
Comment (JRWu) [inline tyk-docs/content/product-stack/tyk-enterprise-developer-portal/release-notes/portal-1.8.4.md]: This formatting block doesn't seem to be rendering correctly

 [PORTAL_SESSION_SECURE]({{ ref “/product-stack/tyk-enterprise-developer-portal/deploy/configuration/#portal_session_secure” }})

File: tyk-docs/content/product-stack/tyk-enterprise-developer-portal/release-notes/portal-1.8.4.md [hunk 19]
```diff
@@ -0,0 +1,89 @@
...
+- Changed the behaviour of removing a developer profile within the developers UI in the admin app. Now, when an admin tries to remove a developer profile and some of their credentials have been removed from the credentials provider, or if the provider itself is down or unreachable, the portal asks the admin if they still want to remove the developer profile by displaying a modal window.
+- Extended the `DELETE /users/:id` API endpoint by adding the [?force]({{< ref "/product-stack/tyk-enterprise-developer-portal/api-documentation/tyk-edp-api" >}}) parameter to force removal of a user even if some of their credentials have been removed from the credentials provider, or if the provider itself is down or unreachable.
```
Comment (JRWu) [inline tyk-docs/content/product-stack/tyk-enterprise-developer-portal/release-notes/portal-1.8.4.md]: is ?force intentional here?

File: tyk-docs/content/product-stack/tyk-enterprise-developer-portal/portal-customisation/customise-user-model.md [hunk 20]
```diff
@@ -0,0 +1,84 @@
...
+menu:
+    main:
+        parent: "Customise the portal"
+weight: 3
+---
+
+{{< note success >}}
+**Tyk Enterprise Developer Portal**
+
+If you are interested in getting access, contact us at [support@tyk.io](<mailto:support@tyk.io?subject=Tyk Enterprise Portal Beta>)
```
Comment (JRWu) [inline tyk-docs/content/product-stack/tyk-enterprise-developer-portal/portal-customisation/customise-user-model.md:15]: Maybe remove the Beta tag here now that the Portal is generally available

File: tyk-docs/content/product-stack/tyk-enterprise-developer-portal/portal-customisation/customise-user-model.md [hunk 21]
```diff
@@ -0,0 +1,84 @@
...
+- **Type of attribute**: The type of data that can be stored in this attribute. You cannot change the value of this field once the attribute is created. The following data types are acceptable:
+  - Boolean (true or false).
+  - Dropdown (a list of values).
+  - String.
+  - Number.
+- **Validation Reg Exp**: A regular expression that is used to validate the value of this field. It is available for the **String** data type only.
+- **Enable validation**: Determines if the portal should apply the regular expression defined in the **Validation Reg Exp** to validate the value of this attribute when creating or updating a user profile. It is available for the **String** data type only.
+- **Dropdown Values**: A list of values for the attribute. It is available for the **Dropdown** data type only.
+- Fields that define the attribute's behavior:
+  - **Write once read many**: Determines if the value of the attribute can be changed after a user profile is created.
```
Comment (JRWu) [inline tyk-docs/content/product-stack/tyk-enterprise-developer-portal/portal-customisation/customise-user-model.md]: This statement is slightly ambiguous- if I check this box does it mean that the field is now write once read many? (i.e. uneditable?)

**Review by JRWu — CHANGES_REQUESTED** [review]
Content looks great to me, just requesting a few minor changes

File: tyk-docs/content/product-stack/tyk-enterprise-developer-portal/release-notes/portal-1.8.4.md (hunk 19 above)
Comment (slega) [reply tyk-docs/content/product-stack/tyk-enterprise-developer-portal/release-notes/portal-1.8.4.md]: Yep.

File: tyk-docs/content/product-stack/tyk-enterprise-developer-portal/portal-customisation/customise-user-model.md (hunk 20 above)
Comment (slega) [reply tyk-docs/content/product-stack/tyk-enterprise-developer-portal/portal-customisation/customise-user-model.md:15]: Well, we will need to update it in other pages as well, so I'm keeping it for now with intention to remove it on all the other pages in H1.

File: tyk-docs/content/product-stack/tyk-enterprise-developer-portal/portal-customisation/customise-user-model.md (hunk 21 above)
Comment (slega) [reply tyk-docs/content/product-stack/tyk-enterprise-developer-portal/portal-customisation/customise-user-model.md]: In this section admins configure behavior of fields, not the values. The values of custom attributes for the user model are editable from the user menu. In other words, to set a value of an attribute for a specific user, you need to go to the Users menu, select the user, and change the required attributes. Write once read many in this case means that you can't do that after a user profile is created.

Comment (slega) [reply tyk-docs/content/product-stack/tyk-enterprise-developer-portal/portal-customisation/customise-user-model.md]: Added that to the doc.

File: tyk-docs/content/product-stack/tyk-enterprise-developer-portal/release-notes/portal-1.8.4.md (hunk 18 above)
Comment (slega) [reply tyk-docs/content/product-stack/tyk-enterprise-developer-portal/release-notes/portal-1.8.4.md]: Fixed

**Review by dcs3spp — APPROVED** [review]
PR LGTM @slega 
Thanks for the peer review @JRWu. Would you be able to approve if changes made by @slega are ok?

**Review by JRWu — APPROVED** [review]
LGTM!

**Pull request merged**